src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -476,19 +476,37 @@ export default class QuestionnaireForm extends Component {
   }
 
   getPractitioner() {
-    if(this.props.cqlPrepoulationResults.BasicPractitionerInfoPrepopulation.OrderingProvider) {
-        return "Practitioner/" +
-        this.props.cqlPrepoulationResults.BasicPractitionerInfoPrepopulation
-          .OrderingProvider.id.value;
+    var p = "Unknown";
+    var requestType = "Unknown";
+    if (this.props.deviceRequest) {
+      requestType = this.props.deviceRequest.resourceType;
+      if (requestType == "DeviceRequest") {
+        p = this.props.deviceRequest.performer.referencee;
+      } else if (requestType == "ServiceRequest") {
+        p = this.props.deviceRequest.performer.reference;
+      } else if (requestType == "MedicationRequest") {
+        p = this.props.deviceRequest.requester.reference;
+      }
     }
-
+    console.log("getPractitioner(): " + requestType + ": " + p);
+    return p;
   }
 
   getPatient() {
-    return "Patient/" + 
-        this.props.cqlPrepoulationResults.BasicPatientInfoPrepopulation
-          .Patient.id.value;
-
+    var p = "Unknown";
+    var requestType = "Unknown";
+    if (this.props.deviceRequest) {
+      requestType = this.props.deviceRequest.resourceType;
```

Comment:
While this works, the patient is also available as from the SMART client.  It could be simpler to pull the patient off of that instead.  Of course, this also works since the device request is required to have a subject.

Reply:
With the SMART client would that require another round trip to the FHIR server? I'm not sure which SMART client you are referring to. If we were to query for the patient that would require knowing which patient and that is all we are really getting here. This method is just to retrieve the Patient ID.